src/eos-metrics-instrumentation.c

```diff
@@ -75,6 +86,94 @@ static gint64 start_time;
 
 static EinsPersistentTally *persistent_tally;
 
+static gboolean
+get_os_version (gchar **name_out,
+                gchar **version_out)
+{
+    g_autoptr(GError) error = NULL;
```

Comment:
For the record, I think they are the best thing since sliced bread, and I would definitely use them for new projects. I'm not thrilled about frequently jumping back and forth between auto pointers and manual memory management in a single code base though. My brain will explode.